### PR TITLE
gh: add pull request template

### PR DESCRIPTION
**Pull request by drahnr**
Adds a pull request template for github, to provide a standard set of contextual information to ease the review process and avoid delays due to information retrieval.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+Leave a message to whoever is going to review this PR.
+Mainly, pointers to review the PR, and how they can test it.
+If things are still WIP or feedback on particular impl details
+are wanted, state them here too.
+-->
+
+## 📜 Checklist
+
+- [ ] Test coverage is excellent and passes
+- [ ] Works when tests are run `--all-features` enabled
```
Comment (cmeister2) [inline .github/pull_request_template.md:44]: tests are run *and* `--all-features` enabled?

As it stands the `test-with-podman` test makes this difficult to properly test. I'd rather have this be a "CI passes" check.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<!--
+Thank you for submitting a PR to the rust rpm implementation!
+-->
+
+# PR Content Desc
```
Comment (cmeister2) [inline .github/pull_request_template.md:5]: Let's splash out and spend the extra few bytes to make this "Description" 🙂

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+- 🩹 Bug Fix
+- 🦚 Feature
+- 📙 Documentation
+- 🦣 Legacy
+- 🪣 Misc
+
+<!---
+Mention the linked issue here.
+This will automatically close the issue once the PR is merged
+and creates a cross reference.
```
Comment (cmeister2) [inline .github/pull_request_template.md:20]: What if there's not a linked issue?

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+Tell the reviewer WHAT changed, and WHY it had to change
+and how the WHAT and the WHY tie together.
+-->
+
+## Notes to reviewer
+
+<!---
+Leave a message to whoever is going to review this PR.
+Mainly, pointers to review the PR, and how they can test it.
+If things are still WIP or feedback on particular impl details
```
Comment (cmeister2) [inline .github/pull_request_template.md:37]: "implementation" - might as well be explicit.

File: .github/pull_request_template.md (hunk 1 above)
Comment (drahnr) [reply .github/pull_request_template.md:44]: The CI part does not work, due to containerization, which last time I checked, does not work.

Comment (dralley) [reply .github/pull_request_template.md:44]:
```suggestion
- [ ] Works when tests are run with `--all-features` enabled
```

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+are wanted, state them here too.
+-->
+
+## 📜 Checklist
+
+- [ ] Test coverage is excellent and passes
```
Comment (dralley) [inline .github/pull_request_template.md:43]: Should we enable codecov reports?  It looks like we collect data with tarpaulin but we don't appear to be doing much with it.

Comment (drahnr) [reply .github/pull_request_template.md:43]: That's a bit out of scope for this PR, and mandates a separate discussion

File: .github/pull_request_template.md (hunk 2 above)
Comment (drahnr) [reply .github/pull_request_template.md:5]: I'll take care of it, it'd be great if you could add suggestions for the edits in future PRs, so I can fix it instantly

Comment (drahnr) [reply .github/pull_request_template.md:5]:
```suggestion
# PR Content Description
```

File: .github/pull_request_template.md (hunk 3 above)
Comment (drahnr) [reply .github/pull_request_template.md:20]: Well, then one should be created, or not linked. I don't think it should contain any possible path of the decision tree, I will apply suggestions to change it though, no strong opinion.

File: .github/pull_request_template.md (hunk 4 above)
Comment (drahnr) [reply .github/pull_request_template.md:37]:
```suggestion
If things are still WIP or feedback on particular implementation details
```

**Review by dralley — APPROVED** [review]

**Comment by dralley** [comment]
I believe the clippy complaints are fixed in one of my PRs already

File: .github/pull_request_template.md (hunk 3 above)
Comment (cmeister2) [reply .github/pull_request_template.md:20]: (I have no strong opinion either, just wanted to raise the point)

**Review by dralley — APPROVED** [review]

**Pull request merged**